pkg/keyidentifier/testdata/make.sh

```diff
@@ -3,97 +3,162 @@
 }
 
-# My openssh is so new, it only makes the new format
-function makeOpensshKey {
+DATA_DIR="specs"
+
+# generate some keys and the corresponding json spec files for testing fingerprints
+function makeOpensshKeyAndSpec {
     type=$1
     bits=$2
-    os="openssh7"
-    dir_fragment="$type/$bits/openssh-new"
-    mkdir -p {encrypted,plaintext}"/$dir_fragment"
-    ssh-keygen -t $type -b $bits  -C "" -f "plaintext/$dir_fragment/seph-$os" -P ""
-    ssh-keygen -t $type -b $bits  -C "" -f "encrypted/$dir_fragment/seph-$os" -P "$(rand)"
-}
-
-makeOpensshKey rsa 1024
-makeOpensshKey rsa 2048
-makeOpensshKey rsa 4096
+    encrypted=$3
+    format="openssh7"
+    source="openssh"
 
-makeOpensshKey dsa 1024
+    keypath="$DATA_DIR/$(rand)"
 
-makeOpensshKey ecdsa 256
-makeOpensshKey ecdsa 521
+    passphrase=""
+    if [ $encrypted == true ]; then
+        passphrase=$(rand)
+    fi
 
-# ed25519 is fixed legth, bits is ignored
-makeOpensshKey ed25519 2048
+    ssh-keygen -t $type -b $bits  -C "" -f $keypath -P "$passphrase"
+
+    cmd="ssh-keygen -t $type -b $bits  -C '' -f $keypath -P $passphrase"
+    fingerprint=$(ssh-keygen -l -f $keypath | awk '{print $2}')
+    md5fingerprint=$(ssh-keygen -l -E md5 -f $keypath | awk '{print $2}' | sed 's/^MD5://')
+
+    cat <<EOF > $keypath.json
+    {
+      "FingerprintSHA256": "$fingerprint",
+      "FingerprintMD5": "$md5fingerprint",
+      "Type": "$type",
+      "Bits": $bits,
+      "Encrypted": $encrypted,
+      "command": "$cmd",
+      "Format": "openssh-new",
+      "Source": "ssh-keygen"
+    }
+EOF
+}
 
 # ssh.com style
 #
 # Note that openssh's ssh-keygen proportes to convert (using `-e`) but
 # empirically this does not work. So we use puttygen (`brew install
 # putty` to generate these)
 
-function makePuttyKeyPuttyFormat {
+# check if puttygen is installed
+hash puttygen 2>/dev/null || \
+    { echo >&2 "puttygen must be installed to generate test data for putty keys. use 'brew install putty' to install puttygen on macos"; exit 1; }
+
+function makePuttyKeyAndSpecFile {
     type=$1
     bits=$2
-    format="putty"
-
-    if [ "$type" == "rsa1" ]; then
-        format="ssh1"
+    format=$3
+    encrypted=$4
+    source="putty"
+
+    putty_format="-$format"
+    if [ "$format" == "putty" ]; then
+        putty_format=""
+        if [ "$type" == "rsa1" ]; then
+            format="ssh1"
+        fi
     fi
 
-    dir_fragment="$type/$bits/$format"
-    mkdir -p {encrypted,plaintext}"/$dir_fragment"
+    keypath="$DATA_DIR/$(rand)"
+
+    passphrase=""
+    if [ $encrypted == true ]; then
+        passphrase=$(rand)
+    fi
 
-    puttygen -t $type -b $bits -C "" -o "plaintext/$dir_fragment/seph-putty" -O private --new-passphrase /dev/null
-    puttygen -t $type -b $bits -C "" -o "encrypted/$dir_fragment/seph-putty" -O private --new-passphrase <(rand)
+    puttygen -t $type -b $bits -C "" -o $keypath -O private$putty_format --new-passphrase <(echo $passphrase)
```

Comment:
It's worth checking that this does what you need. I wonder if that needs to be an `echo -n`

Reply:
it works fine for executing the command, but outputs the file handle (`/dev/fd/63`) for the documentation. which is fine imo.

Reply:
Agree that's fine for the docs. And, I guess the trailing new line doesn't matter here. If the password gets munged, it doesn't matter for this